### PR TITLE
Port to gnuradio 3.8 and python3

### PR DESCRIPTION
**Pull request by bkerler**

### CODE REVIEW

**Comment by argilo** [comment]
Thanks so much for contributing this! I'll have a look over it to see if any changes are needed before merging it.

File: docs/doxygen/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # Copyright 2011 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/CMakeLists.txt]: It looks like the project name is set to gr-foot here and in various other places; this should be changed to gr-flarm.

Comment (argilo) [inline docs/doxygen/CMakeLists.txt]:
```suggestion
# This file is a part of gr-flarm
```

File: docs/doxygen/Doxyfile.in [hunk 2]
```diff
@@ -28,7 +28,7 @@ DOXYFILE_ENCODING      = UTF-8
 # identify the project. Note that if you do not use Doxywizard you need
 # to put quotes around the project name if it contains spaces.
 
-PROJECT_NAME           = "GNU Radio's FLARM Package"
+PROJECT_NAME           = "GNU Radio's FOOT Package"
```
Comment (argilo) [inline docs/doxygen/Doxyfile.in]:
```suggestion
PROJECT_NAME           = "GNU Radio's FLARM Package"
```

File: docs/doxygen/doxyxml/__init__.py [hunk 3]
```diff
@@ -1,7 +1,8 @@
 #
 # Copyright 2010 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/doxyxml/__init__.py]:
```suggestion
# This file is a part of gr-flarm
```

File: docs/doxygen/doxyxml/base.py [hunk 4]
```diff
@@ -1,7 +1,8 @@
 #
 # Copyright 2010 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/doxyxml/base.py]:
```suggestion
# This file is a part of gr-flarm
```

File: docs/doxygen/doxyxml/doxyindex.py [hunk 5]
```diff
@@ -1,7 +1,8 @@
 #
 # Copyright 2010 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/doxyxml/doxyindex.py]:
```suggestion
# This file is a part of gr-flarm
```

File: docs/doxygen/doxyxml/text.py [hunk 6]
```diff
@@ -1,7 +1,8 @@
 #
 # Copyright 2010 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/doxyxml/text.py]:
```suggestion
# This file is a part of gr-flarm
```

File: docs/doxygen/swig_doc.py [hunk 7]
```diff
@@ -1,7 +1,8 @@
 #
-# Copyright 2010,2011 Free Software Foundation, Inc.
+# Copyright 2010-2012 Free Software Foundation, Inc.
 #
-# This file is part of GNU Radio
+# This file was generated by gr_modtool, a tool from the GNU Radio framework
+# This file is a part of gr-foot
```
Comment (argilo) [inline docs/doxygen/swig_doc.py]:
```suggestion
# This file is a part of gr-flarm
```

File: python/packetize.py [hunk 8]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
                 bytes = self.decrypt_packet(in_bytes, key)
                 icao, lat, lon, alt, vs, no_track, stealth, typ, ns, ew, status, unk = self.extract_values(bytes[3:27])
                 if unk in [0x0000, 0x0400, 0x1000, 0x1400]:
-                    print "Time offset: " + str(offset*64),
+                    print("Time offset: " + str(offset*64)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Time offset: " + str(offset*64), end=" ")
```

File: python/packetize.py [hunk 9]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
-                    print "Type: " + str(typ),
-                    print "GPS status: " + str(status),
-                    print "North/South speeds: {0},{1},{2},{3}".format(*ns),
-                    print "East/West speeds: {0},{1},{2},{3}".format(*ew),
-                    print "Unknown: {0:04x}".format(unk),
-                    print "Raw: {0:02x}".format(bytes[6]),
-                    print "{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[7:15]),
-                    print "{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[15:23]),
-                    print "{0:02x}{1:02x}{2:02x}{3:02x}".format(*bytes[23:27]),
+                    print(datetime.utcfromtimestamp(time).isoformat() + 'Z'+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print(datetime.utcfromtimestamp(time).isoformat() + 'Z', end=" ")
```

File: python/packetize.py [hunk 10]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Ch.{0:02}".format(channel)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Ch.{0:02}".format(channel), end=" ")
```

File: python/packetize.py [hunk 11]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("ICAO: " + icao+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("ICAO: " + icao, end=" ")
```

File: python/packetize.py [hunk 12]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Lat: " + str(lat)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Lat: " + str(lat), end=" ")
```

File: python/packetize.py [hunk 13]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Lon: " + str(lon)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Lon: " + str(lon), end=" ")
```

File: python/packetize.py [hunk 14]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Alt: " + str(alt) + "m"+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Alt: " + str(alt) + "m", end=" ")
```

File: python/packetize.py [hunk 15]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("VS: " + str(vs)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("VS: " + str(vs), end=" ")
```

File: python/packetize.py [hunk 16]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("No-track: " + str(no_track)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("No-track: " + str(no_track), end=" ")
```

File: python/packetize.py [hunk 17]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Stealth: " + str(stealth)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Stealth: " + str(stealth), end=" ")
```

File: python/packetize.py [hunk 18]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Type: " + str(typ)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Type: " + str(typ), end=" ")
```

File: python/packetize.py [hunk 19]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("GPS status: " + str(status)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("GPS status: " + str(status), end=" ")
```

File: python/packetize.py [hunk 20]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("North/South speeds: {0},{1},{2},{3}".format(*ns)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("North/South speeds: {0},{1},{2},{3}".format(*ns), end=" ")
```

File: python/packetize.py [hunk 21]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("East/West speeds: {0},{1},{2},{3}".format(*ew)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("East/West speeds: {0},{1},{2},{3}".format(*ew), end=" ")
```

File: python/packetize.py [hunk 22]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Unknown: {0:04x}".format(unk)+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Unknown: {0:04x}".format(unk), end=" ")
```

File: python/packetize.py [hunk 23]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("Raw: {0:02x}".format(bytes[6])+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("Raw: {0:02x}".format(bytes[6]), end=" ")
```

File: python/packetize.py [hunk 24]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[7:15])+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[7:15]), end=" ")
```

File: python/packetize.py [hunk 25]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[15:23])+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("{0:02x}{1:02x}{2:02x}{3:02x}{4:02x}{5:02x}{6:02x}{7:02x}".format(*bytes[15:23]), end=" ")
```

File: python/packetize.py [hunk 26]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
+                    print("{0:02x}{1:02x}{2:02x}{3:02x}".format(*bytes[23:27])+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                    print("{0:02x}{1:02x}{2:02x}{3:02x}".format(*bytes[23:27]), end=" ")
```

File: python/packetize.py [hunk 27]
```diff
@@ -156,43 +156,43 @@ def process_packet(self, channel, bits, time):
...
                     if icao in self.icao_table:
                         reg, typ, tail = self.icao_table[icao]
-                        print "(Reg: " + reg + ", Type: " + typ + ", Tail: " + tail + ")",
-                    print
+                        print("(Reg: " + reg + ", Type: " + typ + ", Tail: " + tail + ")"+" ")
```
Comment (argilo) [inline python/packetize.py]:
```suggestion
                        print("(Reg: " + reg + ", Type: " + typ + ", Tail: " + tail + ")", end=" ")
```

**Comment by argilo** [comment]
I added the suggested fixes and merged. Please let me know how it's working for you. Thanks again!

**Comment by bkerler** [comment]
Yes, seems to work fine ;)

**Pull request merged**